examples/cifar10/train.cpp

```diff
@@ -104,21 +116,75 @@ void train_cifar10(string data_dir_path, double learning_rate, ostream &log) {
                           n_train_epochs, on_enumerate_minibatch,
                           on_enumerate_epoch);
 
-  cout << "end training." << endl;
+  std::cout << "end training." << std::endl;
 
   // test and show results
-  nn.test(test_images, test_labels).print_detail(cout);
-
+  nn.test(test_images, test_labels).print_detail(std::cout);
   // save networks
-  ofstream ofs("cifar-weights");
+  std::ofstream ofs("cifar-weights");
   ofs << nn;
 }
 
+static core::backend_t parse_backend_name(const char *name) {
+  static const char *names[] = {
```

Comment:
@edgarriba I would use `std::array` from that perspective.
http://en.cppreference.com/w/cpp/container/array

But I wouldn't use `std::vector` and `std::string` in here. They aren't fundamentally needed in this routine. `auto` specifier keyword can also be used instead of explicit type declaratios.

if you really want your programs not-succint and resource hogging, you may carelessly persist to use STL containers. I wouldn't be able stop you in that case.